modules/basics/src/main/java/com/opengamma/basics/date/DayCount.java

```diff
@@ -79,15 +80,52 @@ public default double yearFraction(LocalDate firstDate, LocalDate secondDate) {
    * Gets the year fraction between the specified dates.
    * <p>
    * Given two dates, this method returns the fraction of a year between these
-   * dates according to the convention.
+   * dates according to the convention. The dates must be in order.
    * 
    * @param firstDate  the first date
    * @param secondDate  the second date, on or after the first date
    * @param scheduleInfo  the schedule information
-   * @return the year fraction
+   * @return the year fraction, zero or greater
+   * @throws IllegalArgumentException if the dates are not in order
+   * @throws UnsupportedOperationException if the year fraction cannot be obtained
+   */
+  public abstract double yearFraction(LocalDate firstDate, LocalDate secondDate, ScheduleInfo scheduleInfo);
+
+  /**
+   * Gets the relative year fraction between the specified dates.
+   * <p>
+   * Given two dates, this method returns the fraction of a year between these
+   * dates according to the convention. If the first date is after the second date,
+   * the result is calculated by swapping the dates to be in order and negating the result.
```

Comment:
Well, in that case, it needs to explicitly name check the yearFraction method rather than leaving the inference to the reader